kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -320,35 +320,30 @@ impl<'tcx> GotocCtx<'tcx> {
             InstanceDef::DropGlue(_def_id, Some(_)) => {
                 match place_ty.kind() {
                     ty::Dynamic(..) => {
-                        // Virtual drop via a vtable lookup
-                        let trait_fat_ptr = unwrap_or_return_codegen_unimplemented_stmt!(
+                        // Virtual drop via a vtable lookup.
+                        let projection = unwrap_or_return_codegen_unimplemented_stmt!(
                             self,
                             self.codegen_place(place)
-                        )
-                        .fat_ptr_goto_expr
-                        .unwrap();
-                        debug!(?trait_fat_ptr, "codegen_drop: ");
+                        );
+                        debug!(?projection, "codegen_drop");
+                        //  The data comes from the projection.
+                        let self_ref =
+                            if place_ty != pointee_type(self.local_ty(place.local)).unwrap() {
```

Comment:
The check seems to look for a symptom, not the underlying cause. Is there a better check to use here?

Reply:
I can improve the check. No problem!